src/mlpack/methods/ne/tasks.hpp

```diff
@@ -0,0 +1,185 @@
+
+#include <cstddef>
+#include <cmath>
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/performance_functions/mse_function.hpp>
+
+#include "link_gene.hpp"
+#include "neuron_gene.hpp"
+#include "genome.hpp"
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This class defines task xor.
+ */
+template<typename FitnissFunction = ann::MeanSquaredErrorFunction>
+class TaskXor {
+ public:
+  // Evaluate genome's fitness for this task.
+  double EvalFitness(Genome& genome) {
+  	assert(genome.NumInput() == 3); 
+    assert(genome.NumOutput() == 1);
+
+    // Input, output pairs for evaluate fitness.
+  	std::vector<std::vector<double>> inputs;  // TODO: use arma::mat for input.
+  	std::vector<double> input1 = {0, 0, 1};
+  	std::vector<double> input2 = {0, 1, 1};
+  	std::vector<double> input3 = {1, 0, 1};
+  	std::vector<double> input4 = {1, 1, 1};
+  	inputs.push_back(input1);
+  	inputs.push_back(input2);
+  	inputs.push_back(input3);
+  	inputs.push_back(input4);
+
+  	std::vector<double> outputs;
+  	outputs.push_back(1);
+  	outputs.push_back(0);
+  	outputs.push_back(1);
+  	outputs.push_back(0);
+
+  	double fitness = 0;
+  	for (int i=0; i<4; ++i) {
+  		genome.Activate(inputs[i]);
+  		std::vector<double> output;
+      genome.Output(output);
+  	  fitness += pow((output[0] - outputs[i]), 2);
+  	}
+
+    return fitness;  // fitness smaller is better. 0 is best.
+  }
+
+};
+
+/**
+ * This class defines task cart pole balancing.
+ */
+class TaskCartPole {
+ public:
+  // Mass of cart.
+  double mc;
+
+  // Mass of pole.
+  double mp;
+
+  // Gravity.
+  double g;
+
+  // Half length of the pole.
+  double l;
+
+  // Magnitude of control force.
+  double F;
+
+  // Integration time stamp.
+  double tau;
+
+  // Track limits.
+  double track_limit;
+
+  // Angle limits.
+  double theta_limit;
+
+  // Number of trials to test genomes.
+  ssize_t num_trial;
+
+  // Number of steps in each trial.
+  ssize_t num_step;
+
+  // Default constructor.
+  TaskCartPole() {
+    double track_limit = 2.4;
+    double theta_limit = 12 * M_PI / 180.0;
+    double g = 9.81;
+    double mp = 0.1;
+    double mc = 1.0;
+    double l = 0.5;
+    double F = 10.0;
+    double tau = 0.02;
+    ssize_t num_trial = 20;
+    ssize_t num_step = 200;
+  }
+
+  // Parametric constructor.
+  TaskCartPole(double t_mc, double t_mp, double t_g, double t_l, double t_F, 
+               double t_tau, double t_track_limit, double t_theta_limit,
+               ssize_t t_num_trial, ssize_t t_num_step):
+               mc(t_mc), mp(t_mp), g(t_g), l(t_l), F(t_F),
+               tau(t_tau), track_limit(t_track_limit), theta_limit(t_theta_limit),
+               num_trial(t_num_trial), num_step(t_num_step) {}
+
+  // Update status.
+  void Action(double action, double& x, double& x_dot, double& theta, double& theta_dot) {
+    double force = action * F;  // action is -1 or 1. TODO: or continuous???
+    double cos_theta = cos(theta * M_PI / 180.0);
```

Comment:
Maybe I missed something, but the reference code you used:

``
math.cos(theta)
``

 which expects radians as input.